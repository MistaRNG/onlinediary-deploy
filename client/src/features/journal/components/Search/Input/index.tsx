import { InputGroup } from "@blueprintjs/core";
import CancelButton from "./CancelButton";
import React from "react";

interface SearchInputProps {
  search: string;
  changeHandler: (event: React.ChangeEvent<HTMLInputElement>) => void;
  cancel: () => void;
}

const SearchInput: React.FC<SearchInputProps> = ({ search, changeHandler, cancel }) => {
  return (
    <InputGroup
      large={true}
      placeholder="Search..."
      value={search}
      onChange={changeHandler}
      rightElement={<CancelButton search={!!search} cancel={cancel} />}
      className="search-input"
    />
  );
};

export default SearchInput;
