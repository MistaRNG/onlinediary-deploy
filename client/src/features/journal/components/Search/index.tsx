import Input from "./Input";
import useSearch from "common/hooks/useSearch";
import Result from "./Result/ResultList";
import "./Search.css";
import React from "react";

interface SearchProps {}

const Search: React.FC<SearchProps> = () => {
  const { search, changeHandler, cancel, results } = useSearch();

  return (
    <div className="search">
      <Input {...{ search, changeHandler, cancel }} />
      <Result results={results} />
    </div>
  );
};

export default Search;
