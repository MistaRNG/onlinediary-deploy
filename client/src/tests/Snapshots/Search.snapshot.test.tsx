import React from 'react';
import { render } from '@testing-library/react';
import Search from '../../features/journal/components/Search';
import '@testing-library/jest-dom';
import { describe, it, expect } from '@jest/globals';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';

const mockReducer = (state = { data: [], saved: false }, action: any) => state;

const mockStore = configureStore({
  reducer: {
    journals: mockReducer,
  },
});

describe('Search Component Snapshot Tests', () => {
  it('should render the Search component correctly', () => {
    // ACT
    const { asFragment } = render(
      <Provider store={mockStore}>
        <BrowserRouter>
          <Search />
        </BrowserRouter>
      </Provider>
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });
});
