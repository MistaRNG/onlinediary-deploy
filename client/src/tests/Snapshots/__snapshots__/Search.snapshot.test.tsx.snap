// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component Snapshot Tests should render the Search component correctly 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <div
      class="bp4-input-group bp4-large search-input"
    >
      <input
        class="bp4-input"
        placeholder="Search..."
        style="padding-right: 0px;"
        type="text"
        value=""
      />
      <span
        class="bp4-input-action"
      />
    </div>
    <div
      class="search-result"
    >
      No journal is found
    </div>
  </div>
</DocumentFragment>
`;
